contracts/implementations/nodes/AftermarketDevice.sol

```diff
@@ -157,52 +171,136 @@ contract AftermarketDevice is
             );
         }
 
-        // Validate request and transfer funds to foundation
-        // This transfer is at the end of the function to prevent reentrancy
-        _validateMintRequest(msg.sender, devicesAmount);
+        // Validate license and transfer funds to foundation
+        _validateMintRequest(msg.sender, msg.sender, devicesAmount);
     }
 
-    /// @notice Claims the ownership of a list of aftermarket devices to a list of owners
-    /// @dev Caller must have the admin role
-    /// @dev This contract must be approved to spend the tokens in advance
-    /// @param adOwnerPair List of pairs AD-owner
-    function claimAftermarketDeviceBatch(
-        AftermarketDeviceOwnerPair[] calldata adOwnerPair
-    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    /**
+     * @notice Mints aftermarket devices in batch by an authorized address
+     * The manufacturer node owner must grant the minter privilege to the caller
+     * @param manufacturerNode Parent manufacturer node id
+     * @param adInfos List of attribute-info pairs and addresses associated with the AD to be added
+     *  addr -> AD address
+     *  attrInfoPairs[] / attribute
+     *                  \ info
+     */
+    function mintAftermarketDeviceAuthorized(
+        uint256 manufacturerNode,
+        AftermarketDeviceInfos[] calldata adInfos
+    ) external {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
         AftermarketDeviceStorage.Storage storage ads = AftermarketDeviceStorage
             .getStorage();
-        INFT adIdProxy = INFT(ads.idProxyAddress);
+        uint256 devicesAmount = adInfos.length;
+        address adIdProxyAddress = ads.idProxyAddress;
+        INFTMultiPrivilege manufacturerIdProxy = INFTMultiPrivilege(
+            ManufacturerStorage.getStorage().idProxyAddress
+        );
 
-        uint256 aftermarketDeviceNode;
-        address owner;
-        for (uint256 i = 0; i < adOwnerPair.length; i++) {
-            aftermarketDeviceNode = adOwnerPair[i].aftermarketDeviceNodeId;
-            owner = adOwnerPair[i].owner;
+        require(
+            INFT(adIdProxyAddress).isApprovedForAll(msg.sender, address(this)),
+            "Registry must be approved for all"
+        );
+        require(
+            manufacturerIdProxy.exists(manufacturerNode),
+            "Invalid parent node"
+        );
+        require(
+            manufacturerIdProxy.hasPrivilege(
+                manufacturerNode,
+                MANUFACTURER_MINTER_PRIVILEGE,
+                msg.sender
+            ),
+            "Unauthorized"
+        );
 
+        uint256 newTokenId;
+        address deviceAddress;
+
+        for (uint256 i = 0; i < devicesAmount; i++) {
+            newTokenId = INFT(adIdProxyAddress).safeMint(msg.sender);
```

Comment:
I guess we are going to mint these to the manufacturer.

Reply:
Yeah I see